client/me/purchases/remove-purchase/index.jsx

```diff
@@ -0,0 +1,181 @@
+	},
+
+	openDialog( event ) {
+		event.preventDefault();
+
+		this.setState( { isDialogVisible: true } );
+	},
+
+	removePurchase( closeDialog ) {
+		this.setState( { isRemoving: true } );
+
+		const purchase = getPurchase( this.props ),
+			{ selectedSite, user } = this.props;
+
+		removePurchase( purchase.id, user.ID, success => {
+			if ( success ) {
+				if ( isDomainRegistration( purchase ) ) {
+					notices.success(
+						this.translate( `The domain {{em}}%(domain)s{{/em}} was removed from {{em}}%(wpcomUrl)s{{/em}}.`, {
+							args: {
+								domain: purchase.meta,
+								wpcomUrl: selectedSite.wpcom_url
+							},
+							components: { em: <em /> }
+						} ),
+						{ persistent: true }
+					);
+				} else {
+					notices.success(
+						this.translate( `The purchase was removed from {{em}}%(siteSlug)s{{/em}}.`, {
+							args: { siteSlug: selectedSite.slug },
+							components: { em: <em /> }
+						} ),
+						{ persistent: true }
+					);
+				}
+
+				page( purchasePaths.list() );
+			} else {
+				this.setState( { isRemoving: false } );
+
+				closeDialog();
+
+				notices.error( this.props.selectedPurchase.error );
+			}
+		} );
+	},
+
+	renderCard() {
+		const productName = getName( getPurchase( this.props ) );
+
+		return (
+			<CompactCard className="remove-purchase__card" onClick={ this.openDialog }>
+				<a >
+					<Gridicon icon="trash" />
+					{ this.translate( 'Remove %(productName)s', { args: { productName } } ) }
+				</a>
+			</CompactCard>
+		);
+	},
+
+	renderDialog() {
+		const buttons = [ {
+				action: 'cancel',
+				disabled: this.state.isRemoving,
+				label: this.translate( 'Cancel' )
+			},
+			{
+				action: 'remove',
+				disabled: this.state.isRemoving,
+				isPrimary: true,
+				label: this.translate( 'Remove Now' ),
+				onClick: this.removePurchase
+			} ],
+			purchase = getPurchase( this.props ),
+			productName = getName( purchase );
+
+		return (
+			<Dialog
+				buttons={ buttons }
+				className="remove-purchase__dialog"
+				isVisible={ this.state.isDialogVisible }
+				onClose={ this.closeDialog }>
+				<h1>{ this.translate( 'Remove %(productName)s', { args: { productName } } ) }</h1>
+				{ this.renderDialogText() }
+			</Dialog>
+		);
+	},
+
+	renderDialogText() {
+		const purchase = getPurchase( this.props ),
+			productName = getName( purchase );
+
+		if ( isDomainRegistration( purchase ) ) {
+			return (
+				<p>
+					{
+						this.translate( 'This will remove %(domain)s from your account.', {
+							args: { domain: productName },
+							components: { em: <em /> }
+						} )
+					}
+					{ ' ' }
+					{ this.translate( 'By removing, you are canceling the domain registration. This may stop you from using it again, even with another service.' ) }
+				</p>
+			);
+		}
+
+		return (
+			<p>
+				{
+					this.translate( 'Are you sure you want to remove %(productName)s from {{em}}%(domain)s{{/em}}?', {
+						args: { productName, domain: this.props.selectedSite.slug },
```

Comment:
We should probably be consistent and use the same variable names throughout this component. We're using `siteSlug` for the success notice but here we use `domain`.